Mathlib/Topology/Algebra/Module/Action.lean

```diff
@@ -0,0 +1,238 @@
+topology on `A` is `≤` the pullback of the action topology on `M` (because it's the inf of a set
+containing this topology) which is the definition of continuity.
+
+We also deduce that the action topology is a functor from the category of `R`-modules
+(`R` a topological ring) to the category of topological `R`-modules, and it is perhaps
+unsurprising that this is an adjoint to the forgetful functor. Indeed, if `A` is an `R`-module
+and `M` is a topological `R`-module, then the previous paragraph shows that
+the linear maps `A → M` are precisely the continuous linear maps
+from (`A` with its action topology) to `M`, so the action topology is a left adjoint
+to the forgetful functor.
+
+This file develops the theory of the action topology. We prove that the action topology on
+`R` as a module over itself is `R`'s original topology, and that the action topology
+is isomorphism-invariant.
+
+## Main theorems
+
+* `ActionTopology.instSelf : IsActionTopology R R`. The action topology on `R` is `R`'s topology.
+* `ActionTopology.iso [IsActionTopology R A] (e : A ≃L[R] B) : IsActionTopology R B`. If `A` and
+    `B` are `R`-modules with topologies, if `e` is a topological isomorphism between them,
+    and if `A` has the action topology, then `B` has the action topology.
+
+## TODO
+
+Forthcoming PRs from the FLT repo will show that the action topology on a (binary or finite) product
+of modules is the product of the action topologies, and that the action topology on the quotient
+of a module `M` is the quotient topology when `M` is equipped with the action topology.
+
+We will also show the slightly more subtle result that if `M`, `N` and `P` are `R`-modules
+equipped with the action topology and if furthermore `M` is finite as an `R`-module,
+then any bilinear map `M × N → P` is continuous.
+
+As a consequence of this, we deduce that if `R` is a commutative topological ring
+and `A` is an `R`-algebra of finite type as `R`-module, then `A` with its action
+topology becomes a topological ring (i.e., multiplication is continuous).
+
+Other TODOs (not done in the FLT repo):
+
+* The action topology is a functor from the category of `R`-modules
+to the category of topological `R`-modules, and it's an adjoint to the forgetful functor.
+
+-/
+
+section basics
+
+/-
+This section is just boilerplate, defining the action topology and making
+some infrastructure.
+-/
+variable (R : Type*) [TopologicalSpace R] (A : Type*) [Add A] [SMul R A]
+
+/-- The action topology, for a module `A` over a topological ring `R`. It's the finest topology
+making addition and the `R`-action continuous, or equivalently the finest topology making `A`
+into a topological `R`-module. More precisely it's the Inf of the set of
+topologies with these properties; theorems `continuousSMul` and `continuousAdd` show
+that the action topology also has these properties. -/
+abbrev actionTopology : TopologicalSpace A :=
```

Comment:
Should this be in the `Topology` or `TopologicalSpace` namespace? cc @urkud, @YaelDillies .

Reply:
No strong opinion here

Reply:
Yury said (via DM):

Namespace: Either root or TopologicalSpace is fine, because if it's in TopologicalSpace, then you can use letI : TopologicalSpace M := .actionTopology _

Naming: I would call it moduleTopology for the case if someone will separately define some topology associated with a MulAction.